File: packages/gbnf-experiment/data/20260909T130706Z_31ac2789/ported_implementation/src/grammar_graph/get_input_as_code_points.ts
import type { ValidInput } from './grammar_graph_types.ts';

export const getCodePoint = (char: string): number => {
  const codePoint = char.codePointAt(0);
  if (codePoint === undefined) {
    throw new Error(`Could not get code point for character: ${char}`);
  }
  if (!Number.isInteger(codePoint)) {
    throw new Error('code_point must be an integer!');
  }
  return codePoint;
};

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src === 'number') {
    return [src];
  }

  if (Array.isArray(src)) {
    for (const c of src) {
      if (!Number.isInteger(c)) {
        throw new Error(`code_point must be an integer for ${c} if src is a list`);
      }
    }
    return src;
  }

  if (typeof src === 'string') {
    return Array.from(src).map((s) => getCodePoint(s));
  }

  throw new Error(`Invalid input type: ${typeof src}`);
};

export const get_code_point = getCodePoint;
export const get_input_as_code_points = getInputAsCodePoints;

File: proposed-projects/round-trip-experiment/reverse/20260909T001426Z_1c28aa33/20260910T041632Z_e61d8830/ported_implementation/gbnf/grammar_graph/get_input_as_code_points.py
from typing import List

from .grammar_graph_types import ValidInput


def get_code_point(char: str) -> int:
    if len(char) == 0:
        raise ValueError(f"Could not get code point for character: {char}")
    return ord(char[0])


def get_input_as_code_points(src: ValidInput) -> List[int]:
    if isinstance(src, bool):
        raise ValueError(f"Invalid input type: {type(src)}")

    if isinstance(src, int):
        return [src]

    if isinstance(src, list):
        for c in src:
            if not isinstance(c, int) or isinstance(c, bool):
                raise ValueError(
                    f"codePoint must be an integer for {c} if src is a list"
                )
        return src

    if isinstance(src, str):
        return [get_code_point(c) for c in src]

    raise ValueError(f"Invalid input type: {type(src)}")
